### PR TITLE
Update more ug

### LINKED ISSUE #312: [PE-D][Tester D] Person with different name but same phone number is allowed
The `add` command does not check for duplicate phone number. For example, if we input 

* `add r/volunteer h/19 n/James Ho p/22224444 e/jamesho@example.com a/123, Clementi Rd,
1234665 t/friend t/colleague`
* `add r/volunteer h/18 n/Alice p/22224444 e/alice@example.com a/address 1`

, the system will add two person with the same phone number, which is unrealistic.

Expect: The system will raise an error message to inform the phone number is already used by another person.

Maybe it will make the product much better by adding a duplicate check for phone number during adding person. When the user try to add another person but has the same phone number with someone in the list, the system will raise an issue to inform the phone number is already used by other person and do not perform adding.

-------------
Labels: `severity.Medium` `type.FeatureFlaw`
original: jhqiu21/ped#1

### LINKED ISSUE #316: [PE-D][Tester E] User Guide: Donation Amount is made into decimal
Currently, donated amounts are expressed in 1dp ending with "E7". It took a while to understand what that meant and why the value was expressed in decimal place, as there was no explanation found in the UG or DG that this rounding would occur. 
Perhaps, the UG or DG could include an explanation to make it clearer to the user that donated amounts are all expressed in 1 decimal place. 

Steps to recrete: 
```
edit 4 d/12345678
```

-------------
Labels: `type.DocumentationBug` `severity.VeryLow`
original: tingxuanp/ped#2

### LINKED ISSUE #319: [PE-D][Tester A] Wrong error message - delete
**Description**
`delete -1` and `delete 7` ( 7 exceeds the number of list) displays different error messages despite they are both invalid indices

**Steps to reproduce**
Execute command: `delete -1` and `delete 7`

**Expected**
Showing same error message regarding invalid index

**Actual**
Different error messages are shown
**Screenshots**

-------------
Labels: `severity.Low` `type.FeatureFlaw`
original: LynetteLeeSiYing/ped#3

### LINKED ISSUE #323: [PE-D][Tester E] Partnership End Date: invalid date could have more elaboration
The year required to be input for the `PARTNERSHIP_END_DATE` states that the date input need to be a valid date, but I was able to create a date that is not valid (more than 1000 years ago). If this is allowed, perhaps there could be more elaboration on what a 'valid date' means. 

Steps to recreate: 
```
edit 4 r/partner ped/1023-03-02
```

-------------
Labels: `type.FunctionalityBug` `severity.Low`
original: tingxuanp/ped#3

### LINKED ISSUE #327: [PE-D][Tester A] Wrong error message: Edit command
**Description**
Wrong error message for incorrect prefix

**Steps to reproduce**
execute `edit 1 r/partner pred/2022-12-22`

**Expected**
Error message that is relevant

**Actual**
Error message of invalid roles despite having the valid roles inserted

**Screenshots**

-------------
Labels: `severity.Medium` `type.FeatureFlaw`
original: LynetteLeeSiYing/ped#5

### PR DESCRIPTION
**Pull request by zzzlou**
* Fixes #312 
* Fixes #316 
* Fixes #323 
* Fixes #327 
* Fixes #319

### CODE REVIEW

File: docs/UserGuide.md [hunk 1]
```diff
@@ -54,20 +54,37 @@ ContactsForGood (CFG) is a **desktop app for managing contacts, optimized for us
 * Words in `UPPER_CASE` are the parameters to be supplied by the user.<br>
   e.g. in `add n/NAME`, `NAME` is a parameter which can be used as `add n/John Doe`.
 
+* Each parameter is preceded by a **prefix**, which indicates the type of data the parameter represents.  
+  For example:
+    - `n/` is the prefix for `NAME`, as in `add n/John Doe`.
+    - `p/` is the prefix for `PHONE_NUMBER`, as in `add p/12345678`.
+
+* Any unrecognized parameter's prefix will be treated as part of the value for the preceding valid prefix. This means that care should be taken to use only the correct prefixes when entering commands to avoid unexpected behavior.
```
Comment (chongsun2002) [inline docs/UserGuide.md:62]: Genuine question, do you think it will be good to also address the use of / here. For example, what will be the effect of doing something like:

t/NGO target/boss

Will boss be added as a tag? (Can we try this and make it clear here, thanks!)

File: docs/UserGuide.md [hunk 2]
```diff
@@ -54,20 +54,37 @@ ContactsForGood (CFG) is a **desktop app for managing contacts, optimized for us
...
+    - Incorrect: `add n/John Doe xyz/12345678` (Here, `xyz/12345678` will be treated as part of the name value, which
+      will result in an error)
+
 * Items in square brackets are optional.<br>
   e.g `n/NAME [t/TAG]` can be used as `n/John Doe t/friend` or as `n/John Doe`.
 
 * Items with `…`​ after them can be used multiple times including zero times.<br>
   e.g. `[t/TAG]…​` can be used as ` ` (i.e. 0 times), `t/friend`, `t/friend t/family` etc.
 
 * An `INDEX` refers to the person at the index number shown in the **last displayed person list**.<br>
```
Comment (chongsun2002) [inline docs/UserGuide.md:74]: Should we dumb it down for the user even more and include a screenshot of where they can find the index? As it may be unclear to someone who is using the product for the first time.

@seah-minlong Thoughts?

**Review by chongsun2002 — COMMENTED** [review]
I'm ok with the additional details added. Let's think about the suggestions made and let me know what you think!

File: docs/UserGuide.md (hunk 2 above)
Comment (seah-minlong) [reply docs/UserGuide.md:74]: Cannot be bah... numbered list. Considering our ta are ngo administrators too, they should understand when they see the numbers

File: docs/UserGuide.md (hunk 1 above)
Comment (seah-minlong) [reply docs/UserGuide.md:62]: Nope tags must be alphanumeric and have no spacing too

Comment (seah-minlong) [reply docs/UserGuide.md:62]: Hm might need to make this clearer too. Because commands like `add n/Benp/123e/2@gmail.coma/Adres` don't work too even technically, valid prefixes are used. Do we have a definition somewhere that a valid prefix must be <space><prefix>

Comment (chongsun2002) [reply docs/UserGuide.md:62]: > Nope tags must be alphanumeric and have no spacing too. So the example u give cannot be used

what about t/NGOtarget/boss

wonder if this will work. But anyways I think onus is on us to state that it won't

Comment (seah-minlong) [reply docs/UserGuide.md:62]: can't also cus / is not alphanumerical

**Review by chongsun2002 — APPROVED** [review]
LGTM!
